python-sdk/src/astro/databases/google/bigquery.py

```diff
@@ -225,6 +234,32 @@ def is_native_load_file_available(
         location_type = self.NATIVE_PATHS.get(source_file.location.location_type)
         return bool(location_type and file_type)
 
+    def check_schema_autodetection_is_supported(self, source_file: File) -> bool:
+        """
```

Comment:
@sunank200 Can we push these method in database/base.py and just populate `AUTODETECT_SCHEMA_SUPPORTED` and `FILE_PATTERN_BASED_AUTODETECT_SCHEMA_SUPPORTED` in specific databases? 
in databse/base.py can be empty - 
AUTODETECT_SCHEMA_SUPPORTED  = []
FILE_PATTERN_BASED_AUTODETECT_SCHEMA_SUPPORTED = []